File: Java/grains/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    static final BigInteger BIG_INTEGER_2 = BigInteger.valueOf(2);

    public BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
        return BIG_INTEGER_2.pow(square - 1);
    }

    public BigInteger grainsOnBoard() {
        BigInteger total = BigInteger.valueOf(0);
        for (int i = 0; i < 64; i++) {
            total = total.add(BIG_INTEGER_2.pow(i));
        }
        return total;
    }

}

File: Python/grains/grains.py
def square(number):
    if number <= 0 or number > 64:
        raise ValueError('square must be between 1 and 64')
    return 2 ** (number - 1)


def total():
    total = 0
    for num in range(1, 65):
        total += square(num)
    return total
